tests/Fixtures/Codec/TestFileCodec.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace MongoDB\Tests\Fixtures\Codec;
+
+use DateTimeImmutable;
+use MongoDB\BSON\Document;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Codec\DecodeIfSupported;
+use MongoDB\Codec\DocumentCodec;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Tests\Fixtures\Document\TestFile;
+
+final class TestFileCodec implements DocumentCodec
+{
+    use DecodeIfSupported;
+    use EncodeIfSupported;
+
+    public function canDecode($value): bool
+    {
+        return $value instanceof Document;
+    }
+
+    public function decode($value): TestFile
+    {
+        if (! $value instanceof Document) {
+            throw UnsupportedValueException::invalidDecodableValue($value);
+        }
+
+        $fileObject = new TestFile();
+        $fileObject->id = $value->get('_id');
+        $fileObject->length = (int) $value->get('length');
+        $fileObject->chunkSize = (int) $value->get('chunkSize');
```

Comment:
Since we're operating on raw BSON, we may get `Int64` instances, however unlikely they may be (especially for chunkSize). I decided to be prudent and cast these to `int` to avoid that issue.